indexer/modules/custom/pendle/export_pendle_balance_job.py

```diff
@@ -0,0 +1,176 @@
+        self._all_markets_map = {m["market_address"].lower(): m for m in markets}
+
+    def get_filter(self):
+        return [
+            TransactionFilterByLogs(
+                [
+                    TopicSpecification(
+                        addresses=self._all_markets,
+                        topics=[ERC20_TRANSFER_EVENT.get_signature(), redeem_rewards_event.get_signature()],
+                    )
+                ]
+            ),
+        ]
+
+    def _collect(self, **kwargs):
+        pass
+
+    def _process(self, **kwargs):
+        logs: List[Log] = self._data_buff.get(Log.type(), [])
+        if len(logs) == 0:
+            return
+
+        to_requests = set()
+        for log in logs:
+            if log.address.lower() not in self._all_markets:
+                continue
+            if log.topic0 == ERC20_TRANSFER_EVENT.get_signature():
+                to_requests.add((log.address, Web3.to_checksum_address("0x" + log.topic1[26:]), log.block_number))
+                to_requests.add((log.address, Web3.to_checksum_address("0x" + log.topic2[26:]), log.block_number))
+            elif log.topic0 == redeem_rewards_event.get_signature():
+                to_requests.add((log.address, Web3.to_checksum_address("0x" + log.topic1[26:]), log.block_number))
+        if len(to_requests) == 0:
+            return
+
+        calls = []
+        domain_data = []
+        call_to_domain_data = {}
+        for market_address, user_address, block_number in to_requests:
+            if user_address == ZERO_ADDRESS:
+                continue
+            market = self._all_markets_map.get(market_address.lower())
+            if not market:
+                continue
+
+            call_id_prefix = f"{market_address}-{user_address}-{block_number}"
+            domain = PendleUserActiveBalanceD(
+                market_address=market_address,
+                user_address=user_address,
+                block_number=block_number,
+                chain_id=self._chain_id,
+                sy_balance=0,
+                active_balance=0,
+                total_active_supply=0,
+                market_sy_balance=0,
+            )
+            domain_data.append(domain)
+
+            # sy balance
+            c = Call(
+                target=market["sy_address"],
+                function_abi=ERC20_BALANCE_OF_FUNCTION,
+                parameters=[user_address],
+                block_number=block_number,
+            )
+            call_to_domain_data[id(c)] = domain
+            calls.append(c)
+            # user active balance
+            c = Call(
+                target=market_address,
+                function_abi=market_active_balance,
+                parameters=[user_address],
+                block_number=block_number,
+            )
+            call_to_domain_data[id(c)] = domain
+            calls.append(c)
+
+            # market totalActiveSupply
+            c = Call(
```

Comment:
I think you don't need to call this for every user?

Reply:
Each time a user's activityBalance changes, the totalActiveSupply will also be updated, so it needs to be requested. Perhaps the totalActiveSupply field can be separated into its own independent table.